desktop-widgets/charts/chartwidget.cpp

```diff
@@ -0,0 +1,139 @@
+#include "chartwidget.h"
+#include "ui_chartwidget.h"
+#include "qt-models/datestatsmodel.h"
+
+#include <QChart>
+#include <QChartView>
+#include <QVXYModelMapper>
+#include <QBarSeries>
+#include <QStackedBarSeries>
+#include <QBarSet>
+#include <QVBarModelMapper>
+#include <QHBarModelMapper>
+#include <QHeaderView>
+#include <QBarCategoryAxis>
+#include <QValueAxis>
+
+ChartWidget::ChartWidget(QWidget *parent) :
+        QWidget(parent,Qt::Window),
+        ui(new Ui::ChartWidget)
+{
+        ui->setupUi(this);
+        model = new DateStatsTableModel;
+
+        DrawYearChart();
+        DrawMonthChart();
+}
+
+ChartWidget::~ChartWidget()
+{
+        delete ui;
+        delete model;
```

Comment:
Can we please stop using 1980ies style memory management?
Simple way, perhaps longer compilation times:
- make `ui` and `model` a member-object and initialize in initializer list.
More complex way:
- make `ui` and `model` of type `std::unique_ptr<...>`.